app/jobs/data_exports/create_job.rb

```diff
@@ -99,38 +122,75 @@ def create_attachment_manifest_file(attachment)
       attachment_manifest_file
     end
 
-    def write_attachments(sample, project, zip)
+    def write_sample_attachments(sample, project, zip)
       sample.attachments.each do |attachment|
-        next if attachment.metadata.key?('associated_attachment_id') && attachment.metadata['direction'] == 'reverse'
+        directory = "#{project.puid}/#{sample.puid}/#{attachment.puid}/#{attachment.file.filename}"
+        write_attachment(directory, zip, attachment)
+      end
+    end
+
+    # Analysis export specific functions---------------------------------------------------------------------
+    def create_analysis_zip(data_export)
+      new_zip_file = Tempfile.new(binmode: true)
+      ZipKit::Streamer.open(new_zip_file) do |zip|
+        workflow_execution = WorkflowExecution.find(data_export.export_parameters['ids'][0])
+        write_workflow_execution_outputs(workflow_execution, zip)
 
-        write_attachment(project.puid, sample.puid, zip, attachment)
+        sample_workflow_executions = workflow_execution.samples_workflow_executions
+        sample_workflow_executions.each do |swe|
+          write_sample_workflow_execution_outputs(swe, zip) if swe.outputs.count.positive?
+        end
+
+        update_analysis_manifest(workflow_execution, sample_workflow_executions)
+        # Write manifest to file 'manifest.json' and add to zip
+        write_manifest(zip)
+      end
+      new_zip_file
+    end
 
-        next unless attachment.metadata.key?('associated_attachment_id')
+    def update_analysis_manifest(workflow_execution, sample_workflow_executions)
+      add_workflow_execution_outputs_to_manifest(workflow_execution)
 
-        write_attachment(project.puid, sample.puid, zip, attachment.associated_attachment)
+      add_sample_workflow_executions_to_manifest(sample_workflow_executions)
+    end
+
+    def write_workflow_execution_outputs(workflow_execution, zip)
+      workflow_execution.outputs.each do |output|
+        write_attachment(output.file.filename.to_s, zip, output)
       end
     end
 
-    def write_attachment(project_puid, sample_puid, zip, attachment)
-      directory = "#{project_puid}/#{sample_puid}/#{attachment.puid}/#{attachment.file.filename}"
-      zip.write_file(directory) do |writer_for_file|
-        attachment.file.download { |chunk| writer_for_file << chunk }
+    def write_sample_workflow_execution_outputs(swe, zip)
+      sample = Sample.find(swe.sample_id)
```

Comment:
**nitpick (blocking):** This can be changed to `swe.sample` since the `includes` above will all ready have it queried

Reply:
Added in 735c6182225cfb1787142006bed5fcf352e61fb3